evalml/tests/automl_tests/test_dask_engine.py

```diff
@@ -0,0 +1,215 @@
+        same results as simply running the score_pipeline function. """
+        X, y = self.X_y_binary
+        pipeline = TestLRCPipeline({"Logistic Regression Classifier": {"n_jobs": 1}})
+        engine = DaskEngine(client=self.client)
+        objectives = [automl_data.objective]
+
+        pipeline_future = engine.submit_training_job(X=ww.DataTable(X), y=ww.DataColumn(y),
+                                                     automl_data=automl_data, pipeline=pipeline)
+        pipeline = pipeline_future.get_result()
+        pipeline_score_future = engine.submit_scoring_job(X=ww.DataTable(X), y=ww.DataColumn(y),
+                                                          automl_data=automl_data, pipeline=pipeline,
+                                                          objectives=objectives)
+        assert isinstance(pipeline_score_future, DaskComputation)
+        pipeline_score = pipeline_score_future.get_result()
+
+        original_pipeline_score = pipeline.score(X=X, y=y, objectives=objectives)
+
+        assert not np.isnan(pipeline_score["Log Loss Binary"])
+        assert pipeline_score == original_pipeline_score
+
+    def test_submit_scoring_jobs_multiple(self):
+        """ Test that scoring multiple pipelines using the parallel engine produces the
+        same results as the sequential engine. """
+        X, y = self.X_y_binary
+        pipelines = [TestLRCPipeline({"Logistic Regression Classifier": {"n_jobs": 1}}),
+                     TestBaselinePipeline({}),
+                     TestSVMPipeline({})]
+
+        def score_pipelines(pipelines, engine):
+            futures = []
+            for pipeline in pipelines:
+                futures.append(engine.submit_training_job(X=ww.DataTable(X), y=ww.DataColumn(y),
+                                                          automl_data=automl_data, pipeline=pipeline))
+            pipelines = [f.get_result() for f in futures]
+            futures = []
+            for pipeline in pipelines:
+                futures.append(engine.submit_scoring_job(X=ww.DataTable(X), y=ww.DataColumn(y),
+                                                         automl_data=automl_data, pipeline=pipeline,
+                                                         objectives=[automl_data.objective]))
+            results = [f.get_result() for f in futures]
+            return results
+
+        par_eval_results = score_pipelines(pipelines, DaskEngine(client=self.client))
+        par_scores = [s["Log Loss Binary"] for s in par_eval_results]
+
+        seq_eval_results = score_pipelines(pipelines, SequentialEngine())
+        seq_scores = [s["Log Loss Binary"] for s in seq_eval_results]
+
+        # Check there are the proper number of pipelines and all their scores are same.
+        assert len(par_eval_results) == len(pipelines)
+        assert set(par_scores) == set(seq_scores)
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.client.close()
```

Comment:
Very nice

I suggest moving this to be at the top near setup so its obvious to new readers what's going on.